discovery/ecs/ecs_test.go

```diff
@@ -0,0 +1,138 @@
+				},
+				&types.ServiceInstance{
+					Cluster:            "prod-cluster-infra",
+					Service:            "myService",
+					Addr:               "10.0.250.65:30987",
+					Container:          "nginx",
+					ContainerPort:      "8081",
+					ContainerPortProto: "tcp",
+					Image:              "nginx:latest",
+					Labels:             map[string]string{"kind": "front-http"},
+					Tags:               map[string]string{"env": "prod", "kind": "ecs", "cluster": "infra"},
+				},
+			},
+			wantTargets: []string{
+				`{__address__="10.0.250.65:36112", __meta_ecs_cluster="prod-cluster-infra", __meta_ecs_container="myService", __meta_ecs_container_label_kind="main", __meta_ecs_container_label_monitor="true", __meta_ecs_container_port_number="8080", __meta_ecs_container_port_protocol="tcp", __meta_ecs_image="000000000000.dkr.ecr.us-east-1.amazonaws.com/myCompany/myService:29f323e", __meta_ecs_node_tag_cluster="infra", __meta_ecs_node_tag_env="prod", __meta_ecs_node_tag_kind="ecs", __meta_ecs_service="myService"}`,
+				`{__address__="10.0.250.65:24567", __meta_ecs_cluster="prod-cluster-infra", __meta_ecs_container="myService", __meta_ecs_container_label_kind="main", __meta_ecs_container_label_monitor="true", __meta_ecs_container_port_number="1568", __meta_ecs_container_port_protocol="udp", __meta_ecs_image="000000000000.dkr.ecr.us-east-1.amazonaws.com/myCompany/myService:29f323e", __meta_ecs_node_tag_cluster="infra", __meta_ecs_node_tag_env="prod", __meta_ecs_node_tag_kind="ecs", __meta_ecs_service="myService"}`,
+				`{__address__="10.0.250.65:30987", __meta_ecs_cluster="prod-cluster-infra", __meta_ecs_container="nginx", __meta_ecs_container_label_kind="front-http", __meta_ecs_container_port_number="8081", __meta_ecs_container_port_protocol="tcp", __meta_ecs_image="nginx:latest", __meta_ecs_node_tag_cluster="infra", __meta_ecs_node_tag_env="prod", __meta_ecs_node_tag_kind="ecs", __meta_ecs_service="myService"}`,
+			},
+		},
+		{
+			instances:   []*types.ServiceInstance{},
+			wantTargets: []string{},
+			wantError:   true,
+		},
+	}
+
+	for _, test := range tests {
+
+		// Create our mock
+		c := &client.MockRetriever{
+			Instances:   test.instances,
+			ShouldError: test.wantError,
+		}
+
+		d := Discovery{
+			source: "us-west-2",
+			client: c,
+		}
+
+		tgs, err := d.refresh()
+
+		if !test.wantError {
+			if err != nil {
+				t.Errorf("-%+v\n- Refresh shouldn't error, it did: %s", test, err)
+			}
+
+			// Check source
+			if tgs.Source != d.source {
+				t.Errorf("-%+v\n- Source of targets is wrong, want: %s; got: %s", test, d.source, tgs.Source)
+			}
+
+			// Check all the targets are ok
+			if len(test.wantTargets) != len(tgs.Targets) {
+				t.Errorf("-%+v\n- Length of the received target group is not ok, want: %d; got: %d", test, len(test.wantTargets), len(tgs.Targets))
+			}
+
+			for i, got := range tgs.Targets {
+				want := test.wantTargets[i]
+				if want != got.String() {
```

Comment:
Yeah, string comparison is the worst thing, knowing those equality checks, the test should be clearer and more tested, thank you!